code/controllers/DMSDocumentCartController.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+/**
+ * Class DocumentCart_Controller
+ */
+class DMSDocumentCartController extends Controller
+{
+
+    private static $url_handlers = array(
+        '$Action//$ID' => 'handleAction',
+    );
+
+    private static $allowed_actions = array(
+        'DocumentCartForm',
+        'add',
+        'deduct',
+        'remove',
+    );
+
+    /**
+     * See {@link DMSDocumentCart::getItems()}
+     *
+     * @return ArrayList
+     */
+    public function Items()
+    {
+        return ArrayList::create($this->Cart()->getItems());
+    }
+
+    /**
+     * Prepares receiver info for the template
```

Comment:
This is using Zend libraries to do something, please be more specific in your documentation about what it's doing and why